pkg/daemon/ceph/osd/remove.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package osd
+
+import (
+	"fmt"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/daemon/ceph/client"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// RemoveOSDs purges a list of OSDs from the cluster
+func RemoveOSDs(context *clusterd.Context, clusterInfo *client.ClusterInfo, osdsToRemove []string) error {
+
+	// Generate the ceph config for running ceph commands similar to the operator
+	if err := writeCephConfig(context, clusterInfo); err != nil {
+		return errors.Wrap(err, "failed to write the ceph config")
+	}
+
+	// Check if the OSD is down
+	osdDump, err := client.GetOSDDump(context, clusterInfo)
+	if err != nil {
+		return errors.Wrap(err, "failed to get osd dump")
+	}
+
+	for _, osdStatus := range osdDump.OSDs {
+		id64, err := osdStatus.OSD.Int64()
+		if err != nil {
+			continue
+		}
+		id := int(id64)
+
+		logger.Debugf("validating status of osd.%d", id)
```

Comment:
We should only be validating the status of the OSDs that are in the list of OSDs to remove. And if it's in the list, it can be removed as part of this for loop instead of having a separate for loop.